### PR TITLE
Setting network device desired state to be pulled from host adminstate flags

### PR DESCRIPTION
**Pull request by heyvister1**
Applying network device admin state according it device's flags and not operstate
since operstate is not reflecting device's desired state

changing:
- dev_operstate --> dev_adminstate
- vf_operstate --> vf_adminstate
- representor_operstate --> representor_adminstate

```
+ '[' '' == switchdev ']'
+ exec_cmd 'ip link set dev ibp3s0f0 up'
+ debug_print 'Executing command: ip link set dev ibp3s0f0 up'
+ true
+ timestamp_print Executing command: ip link set dev ibp3s0f0 up
++ date +%d-%b-%y_%H:%M:%S
+ date_time_stamp=07-Nov-24_13:17:18
+ msg='[07-Nov-24_13:17:18] Executing command: ip link set dev ibp3s0f0 up'
+ echo '[07-Nov-24_13:17:18] Executing command: ip link set dev ibp3s0f0 up'
+ tee -a /tmp/entrypoint_debug_cmds.log
[07-Nov-24_13:17:18] Executing command: ip link set dev ibp3s0f0 up
++ eval 'ip link set dev ibp3s0f0 up'
+++ ip link set dev ibp3s0f0 up
+ output=

+ exit_code=0
```

### CODE REVIEW

**Comment by rollandf** [comment]
Ignore the lint for now, need to fix the issues that were there before.

File: entrypoint.sh [hunk 1]
```diff
@@ -529,7 +529,8 @@ function find_mlx_devs() {
 
             pci_addr=$(basename $(readlink ${netdev_path}/device)) && [[ -n $pci_addr ]] || return 1
             dev_name=$(basename "$netdev_path") && [[ -n $dev_name ]] || return 1
-            dev_operstate=$(cat "$netdev_path"/operstate) && [[ -n $dev_operstate ]] || return 1
+            dev_operstate_flags=$(( $(cat "$netdev_path"/flags) & 1 )) && [[ -n $dev_operstate_flags ]] || return 1
```
Comment (ykulazhenkov) [inline entrypoint.sh:532]: I don't understand this change. 
Is the patch partially pushed?

We don't need to read operstate file for the device. The operastate file contains operational state of the interface but we need to use administrative state of the interface instead. 

The administrative state of the interface can be controller with `ip link set dev eth0 up` command. 

It is possible to retrieve administrative state from the `flags`. The value of the first bit in the `flags` indicates the administrative state of the interface. 

I propose the following:
1. remove dev_operstate variable
2. add dev_adminstate variable
3. read value of the adminstate  by checking the bit in the `flags`
4. convert the bit value to `up` or `down`
5. use `up` or `down` form of the adminstate in the https://github.com/Mellanox/doca-driver-build/blob/f7f0c925610269c1b7e2c96e62be355f5fad5a8a/entrypoint.sh#L790


We need to fix the logic the same way in all places: for VFs, VF representors and for PFs

Comment (ykulazhenkov) [reply entrypoint.sh:532]:
```
ip -br link show ibp3s0f0

ibp3s0f0         DOWN     <NO-CARRIER,BROADCAST,MULTICAST,UP>
                 ^ this is the operstate                  ^ this is the admin state
```

**Review by ykulazhenkov — APPROVED** [review]
LGTM

**Pull request merged**